### PR TITLE
Bkr fixes

### PR DESCRIPTION
**Pull request by samvarankashyap**
Summary:
- Update schema to allow task addition and method override
- Updated bkr_server module to allow task and method parameters
- Bug fix for incorrect error code after successful provision
- Removed unnecessary debug statements

### CODE REVIEW

File: linchpin/provision/library/bkr_server.py [hunk 1]
```diff
@@ -59,12 +59,26 @@ def provision(self, *args, **kwargs):
...
-            arch = kwargs.get("arch", None)
+            # tasks are list of dictionaries which follows format
+            # for beaker in a box
+            # tasks = [ {arches:[], 'name': '/distribution/utils/dummy'}]
+            # for beaker production
+            # tasks = [ {arches:[], 'name': '/distribution/dummy'}]
+            tasks = kwargs.get("tasks", [{'arches': [], 'name': '/distribution/dummy'}]) 
+            arch = kwargs.get("arch", "x86_64")
             ks_meta = kwargs.get("ks_meta", "")
             method = kwargs.get("method", "nfs")
```
[the remark this answers is not in the document]
Comment (samvarankashyap) [reply linchpin/provision/library/bkr_server.py]: I wanted to change it ""(empty string), however if i do that , everyone who is currently using bkr production server which is in nfs might have to update their topologies to add a attribute method. If that is okay. I would be happy to change it .

File: docs/source/beaker.rst [hunk 2]
```diff
@@ -54,14 +54,16 @@ describes the available recipesets options.
 +==================+============+==========+=========================================+
 | distro           | true       | string   | N/A                                     |
 +------------------+------------+----------+-----------------------------------------+
-| name             | false      | string   | N/A                                     |
+| name             | true       | string   | N/A                                     |
```
Comment (herlo) [inline docs/source/beaker.rst]: Name shouldn't be required here. It's an optional parameter.

File: linchpin/provision/roles/beaker/files/schema.json [hunk 3]
```diff
@@ -21,13 +21,30 @@
                              "type": "dict",
                              "schema": {
                                 "distro": { "type": "string", "required": true },
+                                "name": { "type": "string", "required": true },
```
Comment (herlo) [inline linchpin/provision/roles/beaker/files/schema.json]: Name shouldn't be required. It's an optional parameter.

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**